src/wr_logic_longrange/nodes/obstacle_avoidance.py

```diff
@@ -134,81 +149,97 @@ def update_target(lat, long) -> None:
     @return bool: If we have arrived to the destination, based on our current and target coords
     """
     global target_angle
-    global target_lat
-    global target_long
 
     # Construct the planar target angle relative to east, accounting for curvature
     imu = AngleCalculations(current_lat, current_long,
-                            lat, long)
+                            target_lat, target_long)
 
     target_angle = imu.get_angle() % 360
-    target_lat = lat
-    target_long = long
 
-def update_lidar_data(data: LaserScan) -> None:
-    """
-    Update the stored lidar navigation readings
+    # check if we are close to the target
+    if imu.get_distance() < NAV_THRESH_DISTANCE:
+        return True
+    else:
+        return False
 
-    @param data: Lidar data received
-    """
-    global lidar_data
-    lidar_data = data
 
-def run_navigation() -> bool:
+# Update the robot's navigation and drive it towards the target angle based on our best valley
+def update_navigation(data: LaserScan) -> None:
     """
-    Drive the rover towards the target angle
+    Update the robot's navigation and drive it towards the target angle
 
     @param delta_heading: Relative value of target from robot from -180 to 180. Starting from bottom of robot (Counter-Clockwise)
     @param result: Angle to drive to based on target angle, current heading, and obstacles. Value given as a sector angle with right of robot being 0 (Counterclockwise).
     @param msg: Get the DriveTrainCmd(motor values) relating to the heading of the robot and the resulting best navigation angle
+    @param data: Lidar data received
     """
-    if is_active:
+
+    global frameCount
+
+    # rospy.loginfo(f"target angle: {target_angle}, current heading: {cur_heading}")
+    # calculates average distance 
+    data_avg = sum(cur_range for cur_range in data.ranges) / len(data.ranges)
+
+    #print("Data Avg: " + str(data_avg))
+    # TODO: data threshold might depend of lidar model, double check
+    # Change if units/lidar changes
+    # data_avg is above 0.5 almost always, but result stays the same (?)
+    # TODO (@bennowotny ): This depended on data_avg, why?
+    if True:
         ## Gets best possible angle, considering obstacles
         delta_heading = angle_diff(target_angle, cur_heading)
-        # rospy.logerr(delta_heading)
+        
         result = get_navigation_angle(
             ((((90 + delta_heading) % 360) + 360) % 360)
```

Comment:
Not sure if we need to % 360 then + 360 then % 360 again. Please double check the logic and remove any unnecessary operations.